tools/shoestring/shoestring/healthagents/rest_https_certificate.py

```diff
@@ -24,25 +26,40 @@ def _openssl_run_sclient_verify(hostname, test_args):
 		'localhost:3001'
 	] + test_args, command_input='Q', show_output=False)
 
+	# verify the certificate is valid
+	for line in lines:
+		line = line.strip()
+		if line.startswith('verify error:'):
+			return False, line
+
+	with tempfile.TemporaryDirectory() as temp_directory:
+		temp_file = Path(temp_directory) / 'cert.txt'
+		with open(temp_file, 'wt', encoding='utf-8') as cert_outfile:
+			cert_outfile.writelines(lines)
+
+		lines = openssl_executor.dispatch([
+			'x509',
+			'-inform', 'pem',
+			'-noout',
+			'-text',
+			'-in', temp_file
+		], command_input='Q', show_output=False)
```

Comment:
does `x509` command require `command_input`? (it should be fire and forget, that's why I doubt that)